ambry-protocol/src/test/java/com/github/ambry/protocol/RequestResponseTest.java

```diff
@@ -1011,6 +1009,47 @@ public void ttlUpdateRequestResponseTest() throws IOException {
     }
   }
 
+  /**
+   * Tests for {@link ReplicateBlobRequest} and {@link ReplicateBlobResponse}.
+   * @throws IOException
+   */
+  @Test
+  public void replicateBlobRequestResponseTest() throws IOException {
+    final MockClusterMap clusterMap = new MockClusterMap();
+    final short accountId = Utils.getRandomShort(TestUtils.RANDOM);
+    final short containerId = Utils.getRandomShort(TestUtils.RANDOM);
+    final String clientId = "client";
+    final BlobId id1 = new BlobId(CommonTestUtils.getCurrentBlobIdVersion(), BlobId.BlobIdType.NATIVE,
+        ClusterMap.UNKNOWN_DATACENTER_ID, accountId, containerId,
+        clusterMap.getWritablePartitionIds(MockClusterMap.DEFAULT_PARTITION_CLASS).get(0), false,
+        BlobId.BlobDataType.DATACHUNK);
+    final String sourceHostName = "datacenter1_host1";
+    // 1 ~ 65535 are legit TCP ports. getRandomLong(x, n) returns [0..n)
+    final int sourceHostPort = (int)Utils.getRandomLong(TestUtils.RANDOM, 65535) + 1;
+    final int correlationId = TestUtils.RANDOM.nextInt();
+
+    final ReplicateBlobRequest replicateBlobRequest = new ReplicateBlobRequest(correlationId, clientId, id1, sourceHostName, sourceHostPort);
+    DataInputStream requestStream = serAndPrepForRead(replicateBlobRequest, -1, true);
+    final ReplicateBlobRequest deserializedReplicateBlobRequest = ReplicateBlobRequest.readFrom(requestStream, clusterMap);
+    Assert.assertEquals(correlationId, deserializedReplicateBlobRequest.getCorrelationId());
```

Comment:
Can we have a private method like verifyReplicateBlobRequest and verifyReplicateBlobResponse, and other tests can re-use this verification method?

Reply:
Done.